pkg/controller/job/job_controller.go

```diff
@@ -384,7 +384,8 @@ func (jm *Controller) deletePod(obj interface{}, final bool) {
 		return
 	}
 	job := jm.resolveControllerRef(pod.Namespace, controllerRef)
-	if job == nil {
+	if job == nil || IsJobFinished(job) {
```

Comment:
this doesn't consider deletionTimestamp, right? only the JobComplete or JobFailed status? does the controller always set that status on Jobs that are actively being deleted? for example, a job with a finalizer that prevents removal, and with a deletion timestamp

Reply:
Correct, it doesn't consider the deletionTimestamp, only the job conditions Complete or Failed.

syncJob, the main job-controller worker, continues to synchronize a job until it sets the Complete or Failed condition. So in these cases, we let this worker take care of the finalizers.